stdlib/contrib/bonitoo-io/array/array.flux

```diff
@@ -0,0 +1,106 @@
+// Package array provides functions for interacting with arrays.
+package array
+
+
+import ejson "experimental/json"
+import "json"
+
+// append is a function that appends value to array.
+//
+// ## Parameters
+// - `arr` is the array to operate on.
+// - `v` is the value to append to the array.
+//
+//   Array may be empty. If not, value type must match the type of existing element(s).
+//
+// ## Append to array
+//
+// ```
+// import "contrib/bonitoo-io/array"
+//
+// good = ["foo", "bar"]
+//
+// from(bucket: "my-bucket")
+//   |> range(start: -1h)
+//   |> keep(columns: array.append(arr: ["_time", "_value"], v: good)
+// ```
+builtin append : (arr: [A], v: [A]) => [A]
+
+// map is a function that applies supplied function to each element and returns a new array.
+//
+// ## Parameters
+// - `arr` is the array to operate on.
+// - `fn` is the function to convert items.
+//
+// ## Convert array of ints to array of strings
+//
+// ```
+// import "contrib/bonitoo-io/array"
+//
+// ia = [1, 1]
+//
+// sa = array.map(arr: a, fn: (x) => string(v: x))
+// ```
+builtin map : (arr: [A], fn: (x: A) => B) => [B]
+
+// empty JSON-encoded array as string
+emptyStr = "[]"
+
+// fromStr parses JSON-encoded (as string) array into array.
+//
+// ## Parameters
+// - `arr` is the JSON-encoded (as string) array to operate on.
+//
+// ## Parse simple type array
+//
+// ```
+// import "contrib/bonitoo-io/array"
+//
+// s = "[1, 2, 3]"
+//
+// a = array.fromStr(arr: s)
+// ```
+//
+// ## Parse records array
+//
+// ```
+// import "contrib/bonitoo-io/array"
+//
+// s = "[{\"n\": 1}, {\"n\": 2}, {\"n\": 3}]"
+//
+// a = array.fromStr(arr: s)
+// ```
+fromStr = (arr) => ejson.parse(data: bytes(v: arr))
```

Comment:
Can you explain the user need for these functions? Seems like if you have arrays as JSON strings you would use `json.parse` directly. Why do we want these helper functions in addition?

Reply:
`concatStr` (`appendStr` before) is for use in `reduce`, user cannot aggregate values in an array (in _identity_ record field) because it is not supported Flux column type.
```
import "contrib/bonitoo-io/array"

 from(bucket: "my-bucket")
   |> range(start: -1h)
   |> reduce(
       fn: (r, accumulator) => ({
           sarr: array.concatStr(arr: accumulator.sarr, v: [r._value])
       }),
       identity: {
           sarr: array.emptyStr  // "[]"
       }
   )
   |> map(fn: (r) => ({ r with status:
       http.post(
           url: "http://endpoint:12345/",
           data: json.encode(v: array.fromStr(arr: r.sarr))
       )})
  )
```
`fromStr` is a just helper that decodes such array-string into an array and is used by `concatStr`.

Reply:
Removed `*Str` functions due to perf concerns.